### ISSUE TITLE
Initialize from list of items, like dict and OrderedDict

### ISSUE THREAD

**Issue by MJuddBooth** [drgrib/dotmap#59]
It would be nice if one could do `DotMap((('a', 1), ('b', 2)))` and have it work.  As it is stands now it silently ignores the input.  For now I'm working around it by initializing as`DotMap(OrderedDict(items))`, but I my hacky pull request would probably look something like:

```'python
if isinstance(d, (list, tuple)):
    try:
        d = OrderedDict(d)
    except Exception as e:
        raise ValueError("unable failed to initialize from malformed list of items")
```
but I'm not sure what that would do to your tracking of `id(d)`.

**Comment by MJuddBooth** [comment]
Although I just noticed that  `d = DotMap(OrderedDict(items))` introduces `__get__` and `__set__` into `d._map`.  Only when `_dynamic=True`.  I presume some strange interaction with the magic of OrderedDict.  But avoiding that would suggesting a different fix, where `__call_items` does
```python
if instance(obj, (list, tuple)):
    return iter(obj)
elif ...
```

**Comment by MJuddBooth** [comment]
Although I just noticed that  `d = DotMap(OrderedDict(items))` introduces `__get__` and `__set__` into `d._map`.  Only when `_dynamic=True`.  I presume some strange interaction with the magic of OrderedDict.  But avoiding that would suggesting a different fix, where `__call_items` does
```python
if instance(obj, (list, tuple)):
    return iter(obj)
elif ...
```

**Comment by FelixSchwarz** [comment]
This was fixed in PR #68
